tabbycat/results/scoresheet.py

```diff
@@ -76,6 +79,17 @@ def set_speaker_rank(self, side, position, score):
     def get_speaker_rank(self, side: str, position: int) -> int:
         return self.speaker_ranks[side][position]
 
+    def set_criterion_score(self, side: str, position: int, criterion, score):
+        self.criteria_scores[side][position][criterion] = score
+        weighted_score = float(score) * criterion.weight
+        if self.scores[side][position] is None:
+            self.scores[side][position] = weighted_score
+        else:
+            self.scores[side][position] += weighted_score
```

Comment:
Would this implementation mean that `set_criterion_score` can be set at most once per speaker per criterion, otherwise if (for whatever reason) this is called twice for the same criterion, that score will be double-counted? (In other words, this function is not idempotent?)

If so, I'd either:
1. Rewrite it so that it is idempotent, _i.e._ so that this method _only_ sets the criterion score, and the scores themselves are always computed dynamically. This probably would require spinning out criterion score handling into a new class that does not store `self.scores` or permit `set_score()`, _i.e._ values must only be set via criterion scores. On the one hand, this is probably a good idea(?). On the other hand, it's a lot of work.
2. Check that the criterion score isn't already set, and raise an error (_i.e._ fails fast) if it is not, so that bugs are highlighted quickly rather than becoming time bombs.